### PR TITLE
Simplify conversions around arithmetic operators

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

**Review by axic — APPROVED** [review]
Looks good, at least less operators to implement. Does this have an effect on speed?

File: include/intx/intx.hpp [hunk 1]
```diff
@@ -1785,7 +1775,7 @@ inline void udivrem_knuth(
 }  // namespace internal
 
 template <unsigned M, unsigned N>
-div_result<uint<M>, uint<N>> udivrem(const uint<M>& u, const uint<N>& v) noexcept
+constexpr div_result<uint<M>, uint<N>> udivrem(const uint<M>& u, const uint<N>& v) noexcept
```
Comment (axic) [inline include/intx/intx.hpp:1778]: Was this `constexpr` just missed?

Comment (chfast) [reply include/intx/intx.hpp:1778]: Yes. Maybe we never used it in constexpr context but now compiler is not happy as we removed one level of templates and maybe it is obviously wrong now.

**Comment by chfast** [comment]
No performance changes. In the end we define the same functions just is some other location.

**Pull request merged**